test/e2e/storage/persistent_volumes-local.go

```diff
@@ -518,8 +535,16 @@ func podNodeName(config *localTestConfig, pod *v1.Pod) (string, error) {
 
 // setupLocalVolume setups a directory to user for local PV
 func setupLocalVolume(config *localTestConfig, localVolumeType LocalVolumeType) *localTestVolume {
-	testDirName := "local-volume-test-" + string(uuid.NewUUID())
-	hostDir := filepath.Join(hostBase, testDirName)
+	var hostDir string
+	if localVolumeType == GCELocalSSDVolumeType {
```

Comment:
Maybe the setup and cleanup methods can be abstracted out into an interface for each supported LocalVolumeType

Reply:
Im not sure what you mean by abstracting them out into an interface. Do you mean to have multiple methods for setup and cleanup (one for each volume type)?

Reply:
Yes, I mean define an interface with Setup and Teardown methods.  And have each volume type implement that interface with their own methods.

Reply:
I'm actually not really sure how to do that since localVolumeType aren't actually different "types" but are actually just strings so you can't really distinguish them for interface methods. They are all of type "LocalVolumeType". Any documentation on what you're suggesting here?

Reply:
You could have a map with key = local volume type string, value = setup and teardown methods